### PR TITLE
Index path with original path name

### LINKED ISSUE #5397: readthedocs.org "Search all docs" endpoint forwards to incorrect link (.html missing)
## Details

Hi there, 
I got some trouble regarding the Readthedocs search on your website. 

* Read the Docs project URL: N/A 
* Build URL (if applicable): 3.13.3 ( web version ) 
* Read the Docs username (if applicable): N/A

## How to reproduce 

- visit https://readthedocs.org/
- scroll down to "Search all docs" 
- enter a search phrase in the box. I use :    "RNA-Seq homepage displays"
- click on "let's do this"  
- on result page, click on "Files" 
- click on a result. 

## Expected Result
I expect that I get forward to a URL which shows the result and has the suffix ".html" : 

https://microscope.readthedocs.io/en/latest/content/transcriptomic/NGSProjectRNAseq.html?highlight=getting%20started

## Actual Result

I get forwarded to an Error page saying "SORRY the page does not exist yet" , because the result url is missing the ".html" suffix

Incorrect: https://microscope.readthedocs.io/en/latest/content/transcriptomic/NGSProjectRNAseq?highlight=getting%20started

Correct: NGSProjectRNAseq**.html**?highlight=...
Incorrect forward: NGSProjectRNAseq?highlight=...

### PR DESCRIPTION
**Pull request by stsewd**
Wen we index to es from the ImportedFiles models, we don't save the path from the model, but instead the page name from the fjson file. The page name doesn't include the extension, we used to rely on the doctype in the resolver, but that got removed.

We talked about reindexing all with `path` being the real `path` from the model, but that would require more changes, and we are not sure if that's the solution (removing page name = path).

So, for now I'm just adding a new field with the original path value.

Fix #5397

### CODE REVIEW

**Comment by stsewd** [comment]
This is going to break all projects, and they will need to trigger a build to be fixed... Guessing we don't want that. So, my thoughts:

I think we should serve this file from our servers 
https://github.com/rtfd/readthedocs.org/blob/master/readthedocs/core/static-src/core/js/doc-embed/search.js

so, if we change something in the search api, we have control on how we represen it
Currently search relies in https://github.com/rtfd/readthedocs.org/blob/8258a3c98eae25175cafbdab70ad671cf40445a2/readthedocs/core/static-src/core/js/doc-embed/search.js#L41-L41
but we already know the full file path (with the extension!)
Or we can just keep serving that file from each build, but don't depend on anything else from the local js
like `DOCUMENTATION_OPTIONS`
So, another way to not break existing projects, create a new attribute `doc.docurl`
that returns the full correct link
and keep around `doc.link` with the current old behavior till we figure out how to migrate that

**Comment by ericholscher** [comment]
Yea, I think we should start indexing the full filename in a new field in ES, then add it as a new endpoint in the docsearch API. Our new code can use this, and the old stuff will continue to work.

**Comment by stsewd** [comment]
Let me know if you want me to upload the minified files here

File: readthedocs/search/parse_json.py [hunk 1]
```diff
@@ -59,40 +59,44 @@ def generate_sections_from_pyquery(body):
...
     sections = []
+    path = ''
     title = ''
     body_content = ''
+
     if 'current_page_name' in data:
         path = data['current_page_name']
     else:
         log.info('Unable to index file due to no name %s', filename)
-        return None
```
Comment (stsewd) [inline readthedocs/search/parse_json.py]: If we return none from this function, `get_processed_json` is going to return None too, we always expect a dict from here. And `get_processed_json` has a default https://github.com/stsewd/readthedocs.org/blob/a2e0e3f5e442b0072a4b7cbac9efadbe2a41c224/readthedocs/projects/models.py#L1254-L1261

**Comment by humitos** [comment]
> We already have this information from the model

Could you edit the description of the PR to include a summary of the problem and what's the solution proposed by the PR? It's hard to review without this context to me.

**Comment by stsewd** [comment]
@humitos updated ^

**Comment by stsewd** [comment]
Hold the review for a moment, I think there is a bettter way to put the new field

**Comment by stsewd** [comment]
Done!

File: readthedocs/search/parse_json.py [hunk 2]
```diff
@@ -59,40 +59,37 @@ def generate_sections_from_pyquery(body):
...
+    """Read the fjson file from disk and parse it into a structured dict."""
     try:
-        with codecs.open(filename, encoding='utf-8', mode='r') as f:
+        with codecs.open(fjson_filename, encoding='utf-8', mode='r') as f:
             file_contents = f.read()
     except IOError:
-        log.info('Unable to read file: %s', filename)
-        return None
+        log.info('Unable to read file: %s', fjson_filename)
+        raise
```
[the remark this answers is not in the document]
Comment (ericholscher) [reply readthedocs/search/parse_json.py]: Ah, I see we're catching it at a higher level, 👍

[the remark this answers is not in the document]
Comment (stsewd) [reply readthedocs/search/parse_json.py]: We were returning None, but we expect to always have a dict.

This is caught by https://github.com/stsewd/readthedocs.org/blob/96a85fa8af3cac8b139bdf99598d119eae0e0163/readthedocs/projects/models.py#L1244-L1260

Which returns a default dict

**Comment by stsewd** [comment]
Added to the deploy card that we need to do a full re-index to see this taking effect.

**Pull request merged**